src/function/vector_map_operation.cpp

```diff
@@ -32,5 +35,73 @@ std::unique_ptr<FunctionBindData> MapVectorOperations::bindFunc(
     return std::make_unique<FunctionBindData>(resultType);
 }
 
+vector_operation_definitions MapExtractVectorOperations::getDefinitions() {
+    vector_operation_definitions definitions;
+    definitions.push_back(make_unique<VectorOperationDefinition>(common::MAP_EXTRACT_FUNC_NAME,
+        std::vector<common::LogicalTypeID>{common::LogicalTypeID::MAP, common::LogicalTypeID::ANY},
+        common::LogicalTypeID::VAR_LIST, nullptr, nullptr, bindFunc, false /* isVarLength */));
+    return definitions;
+}
+
+std::unique_ptr<FunctionBindData> MapExtractVectorOperations::bindFunc(
+    const binder::expression_vector& arguments, kuzu::function::FunctionDefinition* definition) {
+    validateKeyType(arguments[0], arguments[1]);
+    auto vectorOperationDefinition = reinterpret_cast<VectorOperationDefinition*>(definition);
+    vectorOperationDefinition->execFunc =
+        VectorListOperations::getBinaryListExecFunc<operation::MapExtract, common::list_entry_t>(
+            arguments[1]->getDataType());
+    auto returnListInfo =
+        std::make_unique<common::VarListTypeInfo>(std::make_unique<common::LogicalType>(
+            *common::MapType::getValueType(&arguments[0]->dataType)));
+    return std::make_unique<FunctionBindData>(
+        common::LogicalType(common::LogicalTypeID::VAR_LIST, std::move(returnListInfo)));
+}
+
+void MapExtractVectorOperations::validateKeyType(std::shared_ptr<binder::Expression> mapExpression,
+    std::shared_ptr<binder::Expression> extractKeyExpression) {
+    auto mapKeyType = common::MapType::getKeyType(&mapExpression->dataType);
+    if (*mapKeyType != extractKeyExpression->dataType) {
+        throw common::RuntimeException("Unmatched map key type and extract key type");
```

Comment:
BinderException. Also I would make it a static function. The principal is, if a function is only used in one cpp file, we should consider make it static and only visible inside that cpp file

Reply:
Made it static function